functions/Measure-DbaDiskSpaceRequirement.ps1

```diff
@@ -0,0 +1,355 @@
+            Will summarize space by ComputerName and MountPoints.
+
+        .NOTES
+            Tags: Migration
+
+            Website: https://dbatools.io
+            Copyright: (C) Chrissy LeMaire, clemaire@gmail.com
+            License: GNU GPL v3 https://opensource.org/licenses/GPL-3.0
+
+        .LINK
+            https://dbatools.io/Measure-DbaDiskSpaceRequirement
+
+        .EXAMPLE
+            Measure-DbaDiskSpaceRequirement -SI INSTANCE1 -SD DB1 -DI INSTANCE2 -Consolidate
+
+            Calculate a simple migration with one database with then same name at destination. In this scenario,
+            no space is required since we will recover 448 KB on destination disk.
+
+            DatabaseName1   DatabaseName2   Name1     Name2   SizeKB1 SizeKB2 DiffKB ComputerName MountPoint
+            -------------   -------------   -----     -----   ------- ------- ------ ------------ ----------
+            DB1             DB1             DB1       DB1        3072   -3000     72 INSTANCE2    D:\
+            DB1             DB1             DB1_log   DB1_log     504   -1024   -520 INSTANCE2    D:\
+
+            ComputerName MountPoint RequiredSpaceKB
+            ------------ ---------- ---------------
+            INSTANCE2    D:\                   -448
+
+        .EXAMPLE
+            @([PSCustomObject]@{Source='SQL1';Destination='SQL2';Database='DB1'},
+              [PSCustomObject]@{Source='SQL1';Destination='SQL2';Database='DB2'}
+            ) | Measure-DbaDiskSpaceRequirement -Consolidate
+
+            Using a PSCustomObject with 2 databases to migrate
+
+        .EXAMPLE
+            Import-Csv -Path .\migration.csv -Delimiter "`t" | Measure-DbaDiskSpaceRequirement -Consolidate
+
+            Using a CSV file. You will need a header in migration.csv "Source<tab>Destination<tab>Database"
+
+        .EXAMPLE
+            Invoke-DbaSqlCmd -SqlInstance DBA -Database Migrations -Query 'select Source,Destination,DatabaseName from refresh.Migrations' `
+                | Measure-DbaDiskSpaceRequirement -Consolidate -Verbose
+
+            Using a SQL table. We are DBA after all!
+    #>
+
+    [CmdletBinding()]
+    Param(
+        [Parameter(Mandatory,ValueFromPipelineByPropertyName)]
+        [Alias('SI','SqlInstance')]
+        [string]$Source,
+        [Parameter(Mandatory,ValueFromPipelineByPropertyName)]
+        [Alias('SD','Database','DatabaseName')]
+        [string]$SourceDatabase,
+        [Alias('SC')]
+        [PSCredential]$SourceSqlCredential,
+        [Parameter(Mandatory,ValueFromPipelineByPropertyName)]
+        [Alias('DI')]
+        [string]$Destination, 
+        [Parameter(Mandatory=$false,ValueFromPipelineByPropertyName)]
+        [Alias('DD')]
+        [string]$DestinationDatabase,
+        [Alias('DC')]
+        [PSCredential]$DestinationSqlCredential,
+        [PSCredential]$Credential, # For Windows access to MountPoints
+        [switch]$Consolidate # Default would be detail only.
```

Comment:
Inline comments can be removed.

Consolidate sounds like a details parameter we used to have and we have opted to stop using this method of controlling what information is output. We set the default view by only allowing a handful of properties to be output, and then if the user wants to see all of it they would pipe the command to `| select *`.

Reply:
Consolidate generate a second output. I don't understand what you ask of me here. But the only viable option would be to ALWAYS return the consolidation with details or with no details. What is your tought on returning 2 tables?

Reply:
You should be returning one object.

The only new column you have in the example for that second object is `RequiredSpaceKB`. This can just be added to the main object, which already has the other properties.